portfolio/src/main/webapp/script.js

```diff
@@ -16,96 +16,133 @@
-        case 'laughing':
-            emotionEl.innerHTML = '&#128516; ';
-            break;
-        case 'surprised':
-            emotionEl.innerHTML = '&#128562; ';
-            break;
-        case 'sad':
-            emotionEl.innerHTML = '&#128532; ';
-            break;
-        default:
-            emotionEl.innerHTML = '&#128522; ';
-    }
-
-    const deleteButton = document.createElement('button');
-    deleteButton.innerHTML = '&#10005;';
-    deleteButton.className = "delete-button";
-    deleteButton.onclick = function() {
-        deleteComments(author, date, text, emotion);
-    }
-
-    containerDiv.className = 'comment-container';
-    containerDiv.appendChild(emotionEl);
-    const authorText = document.createElement('b');
-    authorText.innerText = author;
-    emotionEl.appendChild(authorText);
-    emotionEl.appendChild(deleteButton);
-    const dateNode = document.createElement('i');
-    dateNode.innerText = date;
-    containerDiv.appendChild(dateNode);
-    liElement.appendChild(containerDiv);
-    const textNode = document.createTextNode(text);
-    liElement.appendChild(textNode);
-
-    return liElement;
-}
\ No newline at end of file
+function createListElement(userName, userEmail, date, text, emotion) {
+	const liElement = document.createElement('li');
+
+	const containerDiv = document.createElement('div');
+
+	const emotionEl = document.createElement('div');
+	switch (emotion) {
+		case 'happy':
+			emotionEl.innerHTML = '&#128522; ';
+			break;
+		case 'laughing':
+			emotionEl.innerHTML = '&#128516; ';
+			break;
+		case 'surprised':
+			emotionEl.innerHTML = '&#128562; ';
+			break;
+		case 'sad':
+			emotionEl.innerHTML = '&#128532; ';
+			break;
+		default:
+			emotionEl.innerHTML = '&#128522; ';
+	}
+
+	const deleteButton = document.createElement('button');
+	deleteButton.innerHTML = '&#10005;';
+	deleteButton.className = "delete-button";
+	deleteButton.onclick = function () {
+		deleteComments(author, date, text, emotion);
+	}
+
+	containerDiv.className = 'comment-container';
+	containerDiv.appendChild(emotionEl);
+	const userNameText = document.createElement('b');
+	const userEmailText = document.createElement('i');
+	userNameText.innerText = userName + ' ';
+	userEmailText.innerText = userEmail;
+	emotionEl.appendChild(userNameText);
+	emotionEl.appendChild(userEmailText);
+	emotionEl.appendChild(deleteButton);
+	const dateNode = document.createElement('i');
+	dateNode.innerText = date;
+	containerDiv.appendChild(dateNode);
+	liElement.appendChild(containerDiv);
+	const textNode = document.createTextNode(text);
+	liElement.appendChild(textNode);
+
+	return liElement;
+}
+
+/** 
+ * Checks if the user is logged in and shows comment submission form.
+ */
+function checklogin() {
+	const pageEl = document.getElementById("page");
+	const page = pageEl.value;
+	const queryStr = 'page=' + page;
+	fetch('/user?' + queryStr).
+	then(response => response.json())
+		.then((user) => {
+			if (user.loggedin) {
```

Comment:
Nit: This works, but is really just checking for the existence of `loggedin` property of `user`. The value of the property being `"true"` doesn't get checked.